azurekeyvault-flexvolume/oauth.go

```diff
@@ -191,6 +195,34 @@ func GetServicePrincipalToken(tenantId string, env *azure.Environment, resource
 	return nil, fmt.Errorf("no credentials provided for AAD application %s", aADClientID)
 }
 
+func retryFetchToken(req *http.Request, maxAttempts int) (resp *http.Response, err error) {
+	attempt := 0
+	// Not using exponential backoff logic because the avg time taken by nmi to complete
+	// identity assignment is ~35s. With exponential backoff we might end up waiting
+	// longer than required. Kubelet poll interval is 300ms which is aggressive and results in
+	// a lot of failed volume mount events. This retry will reduce the number of events in the
+	// case when nmi takes longer than usual.
+	delay := time.Duration(7 * time.Second)
+	client := &http.Client{}
+	for attempt < maxAttempts {
+		resp, err = client.Do(req)
+		if err != nil || resp != nil && resp.StatusCode == http.StatusOK {
```

Comment:
Could you please add a comment here on what the logic is doing:
if err != nil, ie. an error has occurred then shouldn't we just continue and go through the retries ? Isn' the only condition we should return back where err != nil  && resp != nil && resp.StatusCode == http.StatusOK ?

Reply:
@kkmsft Added a comment. We should return only when `err==nil && resp != nil && resp.StatusCode == http.StatusOK`. I've updated that condition. For every other scenario we'll retry.